internal/tableland/impl/mesa.go

```diff
@@ -219,6 +224,11 @@ func (t *TablelandMesa) SetController(
 		return tableland.SetControllerResponse{}, fmt.Errorf("chain id %d isn't supported in the validator", chainID)
 	}
 
+	if !stack.AllowTransactionRelay {
+		return tableland.SetControllerResponse{},
+			fmt.Errorf("chain id %d does not suppport relaying of transactions", chainID)
+	}
+
 	tx, err := stack.Registry.SetController(
 		ctx, common.HexToAddress(caller), tableID, common.HexToAddress(req.Controller))
 	if err != nil {
```

Comment:
There's a comment in the `mesa_test.go` below. Mentioning because it might be collapsed.